java/arcs/core/host/EntityHandleManager.kt

```diff
@@ -336,19 +326,21 @@ private class ReadCollectionOperationsImpl<T : Entity>(
     private val name: String,
     private val entitySpec: EntitySpec<T>,
     private val storageProxy: CollectionProxy<RawEntity>,
-    private val dereferencer: Dereferencer<RawEntity>?
+    private val dereferencerFactory: EntityDereferencerFactory
 ) : ReadCollectionOperations<T> {
     override suspend fun size() = fetchAll().size
     override suspend fun isEmpty() = fetchAll().isEmpty()
 
-    private fun Set<RawEntity>.adaptValues() =
-        map { entitySpec.deserialize(it) }.toSet()
+    private fun adaptValues(values: Set<RawEntity>) = values.map {
```

Comment:
I wonder if it's worth using `mapTo(mutableSetOf())` here instead of calling `toSet()` at the end?

Reply:
Done